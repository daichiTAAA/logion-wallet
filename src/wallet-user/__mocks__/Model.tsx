import { TokenizationRequest } from "../../common/types/ModelTypes";
import { CreateTokenRequest } from "../Model";

export async function createTokenRequest(request: CreateTokenRequest): Promise<TokenizationRequest> {
    return Promise.resolve({
        id: "1234",
        legalOfficerAddress: request.legalOfficerAddress,
        requesterAddress: request.requesterAddress,
        requestedTokenName: request.requestedTokenName,
        bars: request.bars,
        status: "PENDING"
    });
}
