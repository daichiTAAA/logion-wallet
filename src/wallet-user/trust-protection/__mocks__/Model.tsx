export function checkActivation() {
    return Promise.resolve();
}
