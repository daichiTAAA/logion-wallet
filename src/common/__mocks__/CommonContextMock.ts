import { TEST_WALLET_USER } from '../../wallet-user/TestData';
import { COLOR_THEME } from '../TestData';
import Addresses, { AccountAddress } from '../types/Addresses';

export let selectAddress = jest.fn();

export let currentAddress = TEST_WALLET_USER;

export function setCurrentAddress(address: string) {
    currentAddress = address;
}

const DEFAULT_ACCOUNT: AccountAddress = {
    name: "name",
    address: TEST_WALLET_USER,
    isLegalOfficer: false,
}

export let addresses: Addresses = {
    currentAddress: DEFAULT_ACCOUNT,
    addresses: [ DEFAULT_ACCOUNT ]
}

export let balances: any = null;

export let transactions: any = null;

export function useCommonContext() {
    return {
        currentAddress,
        selectAddress,
        addresses,
        balances,
        transactions,
        colorTheme: COLOR_THEME,
    };
}

export function setAddresses(value: Addresses) {
    addresses = value;
}

export function setBalances(value: any) {
    balances = value;
}

export function setTransactions(value: any) {
    transactions = value;
}
