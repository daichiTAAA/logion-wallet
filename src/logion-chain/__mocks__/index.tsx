import { Extension } from '../Keys';

const LogionChainContextProvider = (props: any) => null;

let context = {
    apiState: 'CONNECT_INIT',
    injectedAccounts: []
};

export function setContextMock(value: any) {
    context = value;
}

export function useLogionChain() {
    return context;
}

let extensionAvailable = false;

export function setExtensionAvailable(value: boolean) {
    extensionAvailable = value;
}

export function isExtensionAvailable() {
    return extensionAvailable;
}

let recommendedExtensionValue: Extension | null = null;

export function setRecommendedExtension(value: Extension | null) {
    recommendedExtensionValue = value;
}

export function recommendedExtension(): Extension | null {
    return recommendedExtensionValue;
}

export {
    LogionChainContextProvider,
};
