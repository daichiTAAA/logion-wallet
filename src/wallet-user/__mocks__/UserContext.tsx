import {DEFAULT_LEGAL_OFFICER, TokenizationRequest} from "../../legal-officer/Types";
import {TEST_WALLET_USER} from "../Model.test";
import {ApiPromise} from '@polkadot/api';
import {ProtectionRequest} from "../../legal-officer/Types";
import { DARK_MODE } from '../Types';

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export let pendingTokenizationRequests: TokenizationRequest[] | null = null;

export let acceptedTokenizationRequests: TokenizationRequest[] | null = null;

export let rejectedTokenizationRequests: TokenizationRequest[] | null = null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = () => null;

export let pendingProtectionRequests: ProtectionRequest[] | null = null;

export let acceptedProtectionRequests: ProtectionRequest[] | null = null;

export let api = new ApiPromise();

export let recoveryConfig = null;

export let setUserAddress = jest.fn();

export let userAddress = TEST_WALLET_USER;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress,
        createTokenRequest,
        createdTokenRequest,
        pendingTokenizationRequests,
        acceptedTokenizationRequests,
        rejectedTokenizationRequests,
        refreshRequests,
        createProtectionRequest,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        recoveryConfig,
        api,
        setUserAddress,
        colorTheme: DARK_MODE,
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}

export function setPendingRequests(requests: TokenizationRequest[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectedRequests(requests: TokenizationRequest[]) {
    rejectedTokenizationRequests = requests;
}

export function setAcceptedRequests(requests: TokenizationRequest[]) {
    acceptedTokenizationRequests = requests;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setPendingProtectionRequests(requests: ProtectionRequest[] | null) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: ProtectionRequest[] | null) {
    acceptedProtectionRequests = requests;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setRecoveryConfig(config: any) {
    recoveryConfig = config;
}

export function UserContextProvider() {
    return (
        <div></div>
    );
}

export function setContextUserAddress(address: string) {
    userAddress = address;
}