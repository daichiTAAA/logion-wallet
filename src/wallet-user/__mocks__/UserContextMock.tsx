import {DEFAULT_LEGAL_OFFICER} from "../../common/types/LegalOfficer";
import { TEST_WALLET_USER } from '../TestData';
import {ApiPromise} from '@polkadot/api';
import {TokenizationRequest, ProtectionRequest} from "../../common/types/ModelTypes";
import { DARK_MODE } from '../Types';

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export let pendingTokenizationRequests: TokenizationRequest[] | null = null;

export let tokenizationRequestsHistory: TokenizationRequest[] | null = null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = () => null;

export let pendingProtectionRequests: ProtectionRequest[] | null = null;

export let acceptedProtectionRequests: ProtectionRequest[] | null = null;

export let api = new ApiPromise();

export let recoveryConfig = {isEmpty: true};

export let setUserAddress = jest.fn();

export let userAddress = TEST_WALLET_USER;

export let recoveredAddress: string | null = null;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress,
        createTokenRequest,
        createdTokenRequest,
        pendingTokenizationRequests,
        tokenizationRequestsHistory,
        refreshRequests,
        createProtectionRequest,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        recoveryConfig,
        api,
        setUserAddress,
        colorTheme: DARK_MODE,
        recoveredAddress,
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}

export function setPendingRequests(requests: TokenizationRequest[]) {
    pendingTokenizationRequests = requests;
}

export function setTokenizationRequestsHistory(requests: TokenizationRequest[]) {
    tokenizationRequestsHistory = requests;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setPendingProtectionRequests(requests: ProtectionRequest[] | null) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: ProtectionRequest[] | null) {
    acceptedProtectionRequests = requests;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setRecoveryConfig(config: any) {
    recoveryConfig = config;
}

export function UserContextProvider() {
    return (
        <div></div>
    );
}

export function setContextUserAddress(address: string) {
    userAddress = address;
}

export function setRecoveredAddress(value: string) {
    recoveredAddress = value;
}
