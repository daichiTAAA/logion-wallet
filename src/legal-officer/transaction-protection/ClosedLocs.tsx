import React from 'react';

import { useCommonContext } from '../../common/CommonContext';
import Table, { Cell, EmptyTableMessage, DateTimeCell, ActionCell } from '../../common/Table';
import LocStatusCell from '../../common/LocStatusCell';
import LocIdCell from '../../common/LocIdCell';
import UserIdentityNameCell from '../../common/UserIdentityNameCell';
import Button from "../../common/Button";
import { useNavigate } from "react-router-dom";
import { transactionLocDetailsPath, identityLocDetailsPath } from "../LegalOfficerPaths";
import ButtonGroup from "../../common/ButtonGroup";
import { LocType, IdentityLocType } from '../../logion-chain/Types';

export interface Props {
    locType: LocType;
    identityLocType?: IdentityLocType;
}

export default function ClosedLocs(props: Props) {
    const { closedLocRequests, closedIdentityLocsByType } = useCommonContext();
    const navigate = useNavigate();

    if (closedLocRequests === null || closedIdentityLocsByType === null) {
        return null;
    }

    const requests = props.locType === 'Transaction' ? closedLocRequests : closedIdentityLocsByType[props.identityLocType!];

    return (
        <Table
            columns={[
                {
                    "header": "Requester",
                    render: request => <UserIdentityNameCell userIdentity={ request.userIdentity }/>,
                    align: 'left',
                },
                {
                    "header": "Description",
                    render: request => <Cell content={ request.description } overflowing tooltipId='description-tooltip' />,
                    align: 'left',
                },
                {
                    header: "Status",
                    render: request => <LocStatusCell status={ request.status }/>,
                    width: "140px",
                },
                {
                    header: "LOC ID",
                    render: request => <LocIdCell status={ request.status } id={ request.id } />,
                    align: "left",
                },
                {
                    header: "Creation date",
                    render: request => <DateTimeCell dateTime={ request.createdOn || null } />,
                    width: '200px',
                    align: 'center',
                },
                {
                    header: "Closing date",
                    render: request => <DateTimeCell dateTime={ request.closedOn || null } />,
                    width: '200px',
                    align: 'center',
                },
                {
                    header: "Action",
                    render: request =>
                        <ActionCell>
                            <ButtonGroup>
                                <Button onClick={ () => navigate(request.locType === 'Transaction' ? transactionLocDetailsPath(request.id) : identityLocDetailsPath(request.id)) }>View</Button>
                            </ButtonGroup>
                        </ActionCell>
                    ,
                    width: '200px',
                    align: 'center',
                }
            ] }
            data={ requests.map(requestAndLoc => requestAndLoc.request) }
            renderEmpty={ () => <EmptyTableMessage>No LOCs</EmptyTableMessage> }
        />
    );
}
