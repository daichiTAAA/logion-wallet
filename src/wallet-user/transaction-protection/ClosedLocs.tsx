import React from 'react';

import { useCommonContext } from '../../common/CommonContext';
import Table, { Cell, EmptyTableMessage, DateTimeCell, ActionCell } from '../../common/Table';
import LocStatusCell from '../../common/LocStatusCell';
import LocIdCell from '../../common/LocIdCell';
import LegalOfficerName from '../../common/LegalOfficerNameCell';
import ButtonGroup from "../../common/ButtonGroup";
import Button from "../../common/Button";
import { fullCertificateUrl } from "../../PublicPaths";
import { UUID } from "../../logion-chain/UUID";
import Loader from '../../common/Loader';

export default function ClosedLocs() {
    const { closedLocRequests } = useCommonContext();

    if(closedLocRequests === null) {
        return <Loader />;
    }

    return (
        <Table
            columns={[
                {
                    "header": "Legal Officer",
                    render: requestAndLoc => <LegalOfficerName address={ requestAndLoc.request.ownerAddress } />,
                    align: 'left',
                },
                {
                    "header": "Description",
                    render: requestAndLoc => <Cell content={ requestAndLoc.request.description } />,
                    align: 'left',
                },
                {
                    header: "Status",
                    render: requestAndLoc => <LocStatusCell status={ requestAndLoc.request.status }/>,
                    width: "140px",
                },
                {
                    header: "LOC ID",
                    render: requestAndLoc => <LocIdCell status={ requestAndLoc.request.status } id={ requestAndLoc.request.id }/>,
                    align: "left",
                },
                {
                    "header": "Creation date",
                    render: requestAndLoc => <DateTimeCell dateTime={ requestAndLoc.request.createdOn || null } />,
                    width: '200px',
                    align: 'center',
                },
                {
                    header: "Closing date",
                    render: requestAndLoc => <DateTimeCell dateTime={ requestAndLoc.request.closedOn || null } />,
                    width: '200px',
                    align: 'center',
                },
                {
                    header: "Action",
                    render: requestAndLoc =>
                        <ActionCell>
                            <ButtonGroup>
                                <Button onClick={ () => window.open(fullCertificateUrl(new UUID(requestAndLoc.request.id))) }>Certificate</Button>
                            </ButtonGroup>
                        </ActionCell>
                    ,
                    width: '200px',
                    align: 'center',
                },
            ]}
            data={ closedLocRequests }
            renderEmpty={ () => <EmptyTableMessage>No LOCs</EmptyTableMessage>}
        />
    );
}
