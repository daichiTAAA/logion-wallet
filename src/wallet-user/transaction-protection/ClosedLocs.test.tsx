jest.mock('../../common/CommonContext');

import { shallowRender } from '../../tests';
import ClosedLocs from './ClosedLocs';
import { setClosedLocRequests } from '../../common/__mocks__/CommonContextMock';

test("Renders null with no data", () => {
    const tree = shallowRender(<ClosedLocs />);
    expect(tree).toMatchSnapshot();
});

test("Renders pending requests", () => {
    setClosedLocRequests([
        {
            request: {
                id: "1",
                ownerAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
                requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
                description: "LOC description",
                status: "CLOSED"
            }
        }
    ]);
    const tree = shallowRender(<ClosedLocs/>);
    expect(tree).toMatchSnapshot();
});
