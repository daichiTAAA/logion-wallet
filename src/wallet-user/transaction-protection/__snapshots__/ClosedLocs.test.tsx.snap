// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders null with no data 1`] = `null`;

exports[`Renders pending requests 1`] = `
<Table
  columns={
    Array [
      Object {
        "align": "left",
        "header": "Legal Officer",
        "render": [Function],
      },
      Object {
        "align": "left",
        "header": "Description",
        "render": [Function],
      },
      Object {
        "header": "Status",
        "render": [Function],
        "width": "140px",
      },
      Object {
        "align": "left",
        "header": "LOC ID",
        "render": [Function],
      },
      Object {
        "align": "center",
        "header": "Creation date",
        "render": [Function],
        "width": "200px",
      },
      Object {
        "align": "center",
        "header": "Closing date",
        "render": [Function],
        "width": "200px",
      },
      Object {
        "align": "center",
        "header": "Action",
        "render": [Function],
        "width": "200px",
      },
    ]
  }
  data={
    Array [
      Object {
        "request": Object {
          "description": "LOC description",
          "id": "1",
          "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
          "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
          "status": "CLOSED",
        },
      },
    ]
  }
  renderEmpty={[Function]}
/>
`;
